requirements-base.txt

```diff
@@ -63,23 +44,44 @@ redis-py-cluster==1.3.4
 redis>=2.10.3,<2.10.6
 requests-oauthlib==1.2.0
 requests[security]>=2.20.0,<2.21.0
-selenium==3.141.0
 sentry-relay>=0.5.2,<0.6.0
 sentry-sdk>=0.13.5
-setproctitle>=1.1.7,<1.2.0
 simplejson>=3.2.0,<3.9.0
 six>=1.10.0,<1.11.0
 sqlparse>=0.2.0,<0.3.0
-statsd>=3.1.0,<3.2.0
 strict-rfc3339>=0.7
 structlog==16.1.0
 symbolic>=7.1.0,<8.0.0
 toronado>=0.0.11,<0.1.0
 ua-parser>=0.6.1,<0.8.0
-# for bitbucket client
 unidiff>=0.5.4
 urllib3==1.24.2
 uwsgi>2.0.0,<2.1.0
 
+# not directly used, but provides a speedup for redis
+hiredis>=0.1.0,<0.2.0
+
+# not directly used, but pinned for at least semaphore/symbolic
+cffi>=1.11.5,<2.0
+
+# not directly used, but pinned for toronado
+cssutils==1.0.2
+
+# not directly used, but needed
+setproctitle>=1.1.7,<1.2.0
+
 # sentry-plugins specific dependencies
 phabricator>=0.6.0,<1.0
+
+# test dependencies, but unable to move to requirements-test until sentry.utils.pytest and similar are moved to tests/
+mock==3.0.5 ; python_version < "3.3"
+percy>=1.1.2
+selenium==3.141.0
+
+# optional
```

Comment:
Unrelated, these aren't really optional anymore. This is kind of a relic of the past, but it's easier if we just mandate everything be installed for all backends. Even if they're unused, there's no harm in installing these since they'd need to be configured anyways.

Same story for plugins and why we began to vendor them. It's too much work to strictly consider them optional.